File: app/src/main/java/cn/arsenals/osarsenals/utils/Alog.java
package cn.arsenals.osarsenals.utils;

import android.util.Log;

public class Alog {
    private static String convertToAlogTag(String tag) {
        return "ARSENALS_$tag";
    }

    public static void verbose(String tag, String msg) {
        Log.v(convertToAlogTag(tag), msg);
    }

    public static void verbose(String tag, String msg, Throwable tr) {
        Log.v(convertToAlogTag(tag), msg, tr);
    }

    public static void debug(String tag, String msg) {
        Log.d(convertToAlogTag(tag), msg);
    }

    public static void debug(String tag, String msg, Throwable tr) {
        Log.d(convertToAlogTag(tag), msg, tr);
    }

    public static void info(String tag, String msg) {
        Log.i(convertToAlogTag(tag), msg);
    }

    public static void info(String tag, String msg, Throwable tr) {
        Log.i(convertToAlogTag(tag), msg, tr);
    }

    public static void warn(String tag, String msg) {
        Log.w(convertToAlogTag(tag), msg);
    }

    public static void warn(String tag, String msg, Throwable tr) {
        Log.w(convertToAlogTag(tag), msg, tr);
    }

    public static void error(String tag, String msg) {
        Log.e(convertToAlogTag(tag), msg);
    }

    public static void error(String tag, String msg, Throwable tr) {
        Log.e(convertToAlogTag(tag), msg, tr);
    }

    public static void wtf(String tag, String msg) {
        Log.wtf(convertToAlogTag(tag), msg);
    }

    public static void wtf(String tag, Throwable tr) {
        Log.wtf(convertToAlogTag(tag), tr);
    }

    public static void wtf(String tag, String msg, Throwable tr) {
        Log.wtf(convertToAlogTag(tag), msg, tr);
    }
}

File: app/src/main/java/cn/arsneals/osarsenals/utils/Alog.kt
package cn.arsneals.osarsenals.utils

import android.util.Log

class Alog {
    companion object {
        private fun convertToAlogTag(tag: String): String {
            return "ARSENALS_$tag"
        }

        fun verbose(tag: String, msg: String) {
            Log.v(convertToAlogTag(tag), msg)
        }

        fun verbose(tag: String, msg: String, tr: Throwable?) {
            Log.v(convertToAlogTag(tag), msg, tr)
        }

        fun debug(tag: String, msg: String) {
            Log.d(convertToAlogTag(tag), msg)
        }

        fun debug(tag: String, msg: String, tr: Throwable) {
            Log.d(convertToAlogTag(tag), msg, tr)
        }

        fun info(tag: String, msg: String) {
            Log.i(convertToAlogTag(tag), msg)
        }

        fun info(tag: String, msg: String, tr: Throwable?) {
            Log.i(convertToAlogTag(tag), msg, tr)
        }

        fun warn(tag: String, msg: String) {
            Log.w(convertToAlogTag(tag), msg)
        }

        fun warn(tag: String, msg: String, tr: Throwable?) {
            Log.w(convertToAlogTag(tag), msg, tr)
        }

        fun error(tag: String, msg: String) {
            Log.e(convertToAlogTag(tag), msg)
        }

        fun error(tag: String, msg: String?, tr: Throwable) {
            Log.e(convertToAlogTag(tag), msg, tr)
        }

        fun wtf(tag: String, msg: String?) {
            Log.wtf(convertToAlogTag(tag), msg)
        }

        fun wtf(tag: String, tr: Throwable) {
            Log.wtf(convertToAlogTag(tag), tr)
        }

        fun wtf(tag: String, msg: String?, tr: Throwable) {
            Log.wtf(convertToAlogTag(tag), msg, tr)
        }
    }
}